sources/web/datalab/polymer/components/datalab-editor/datalab-editor.ts

```diff
@@ -161,38 +161,39 @@ class DatalabEditorElement extends Polymer.Element {
                   this._file = model;
                   return this.dispatchEvent(new NotificationEvent('Saved.'));
                 });
+                // TODO: Handle save failure here.
             }
           }
 
           return Promise.resolve(null);
         });
-    }
-
-    // If _file is defined, we're saving to an existing file
-    if (this._file) {
-      const filePath = this._file.path;
-      const dirPath = filePath.substr(0, filePath.lastIndexOf(this._file.name));
-
-      const model: JupyterFile = {
-        content: this._editor.doc.getValue(),
-        created: this._file.created,
-        format: this._file.format,
-        last_modified: new Date().toISOString(),
-        mimetype: this._file.mimetype,
-        name: this._file.name,
-        path: dirPath,
-        type: this._file.type,
-        writable: this._file.writable,
-      };
-
-      return ApiManager.saveJupyterFile(model)
-        .then(() => {
-          this._file = model;
-          return this.dispatchEvent(new NotificationEvent('Saved.'));
-        });
-      // TODO: Handle save failure here.
     } else {
-      return Promise.resolve(null);
+    // If _file is defined, we're saving to an existing file
+      if (this._file) {
```

Comment:
This if-statement is redundant, it is testing the same condition as the else-clause on line 170. The return statement on line 195 is unreachable code.